### PR TITLE
Support math expressions in typst style

### PR DESCRIPTION
**Pull request by Lambdaris**
[Typst](https://typst.app/) is a new markup-based typesetting system and has more consistent and expressive styling system compared with LaTeX. 
From my experience typst style is more comfortable in informal scientific writing such as blog.  [Here](https://typst.app/docs/tutorial/writing-in-typst/#maths) are some examples.

This pr allow users to write typst style mathmatical expressions. If set `math=true`  or `math="KaTeX"` katex is still enabled while if `math=typst` is set a render based on [wypst](https://github.com/0xpapercut/wypst) will take over the maths. The two unreadable files under `static/js` are compiled from [here](https://github.com/Lambdaris/typst-auto-render).

Besides a minor commit loading `data.toml` from section directories is added, by which two or more project-like sections could coexist.

At last, thanks for your pretty themes!

### CODE REVIEW

File: static/js/typst-auto-render.js
Comment (isunjn) [inline static/js/typst-auto-render.js:1]: this lib is using some not-well-supported js features, such as `arr.at(-1)`, maybe you should transpile it to a lower js version.

also seems like your parse function is a little buggy:

```
extractMathText('$ \\$ $')

(3) [{…}, {…}, {…}]
{type: 'text', data: ''}
{type: 'math', data: '\\', rawData: '$ \\$', display: false}
{type: 'text', data: ' $'}
```

**Review by isunjn — CHANGES_REQUESTED** [review]

File: static/js/typst-auto-render.js
Comment (Lambdaris) [reply static/js/typst-auto-render.js:1]: Thanks for so careful reviews. All others are done and this bug is fixed at [5ed1ac](https://github.com/Lambdaris/typst-auto-render/commit/5ed1ac53ec769ac900d8d78de5482147848188ec). 

Besides, sorry for I'm unfamiliar with web. The `at` method is introduced by `String::at` and I have no idea which version to transpile (MDN does not show hints like `Array::at`). Could you specify the js version?

Comment (isunjn) [reply static/js/typst-auto-render.js:1]: check <https://www.npmjs.com/package/babel-loader>, the default option is just fine

File: templates/projects.html [hunk 1]
```diff
@@ -11,7 +11,7 @@
 {% include "_header.html" %}
 <div id="wrapper">
   <main>
-    {% set data = load_data(path="content/projects/data.toml", format="toml") %}
+    {% set data = load_data(path="content/" ~ section.path ~ "/data.toml", format="toml") %}
```
Comment (isunjn) [inline templates/projects.html:14]: section.path is '/projects/', with slashes

File: static/js/typst-auto-render.js
Comment (isunjn) [reply static/js/typst-auto-render.js:1]: my bad, this actualy needs a 'polyfill', babel itself won't do it.

anyway, i'm merging this :)

**Pull request merged**